.github/workflows/tests.yml

```diff
@@ -2,106 +2,39 @@ name: Tests
 
 on: [push, pull_request]
 
-env:
-  python3-8-image: ubuntu/chiselled-python:3.8-test
-  ubuntu-release: 20.04
-
 jobs:
   test:
     name: Build and test
     runs-on: ubuntu-latest
-    outputs:
-      test_status: ${{ steps.set-outputs.outputs.status }}
     strategy:
       fail-fast: false
       matrix:
-        arch: ["amd64", "arm", "arm64", "s390x", "ppc64le"]
+        version: ["3.8"]
 
     steps:
       - uses: actions/checkout@v4
 
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v3
-
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v3
-
-      - name: Build the ${{ env.python3-8-image }} image on ${{ matrix.arch }}
-        run: |
-          docker buildx build \
-            --tag ${{ env.python3-8-image }} \
-            --file python3.8/Dockerfile.${{ env.ubuntu-release }} \
-            --platform linux/${{ matrix.arch }} \
-            --output type=oci,dest="chiselled-python.${{ matrix.arch }}.tar" \
-            python3.8
+      - name: Build python${{ matrix.version}} rock
+        id: build
+        uses: canonical/craft-actions/rockcraft-pack@main
+        with:
+          path: python${{ matrix.version }}/
+          verbosity: debug
+          # we need >= v1.2.0 for running the rock properly
+          revision: 1552
 
       - name: Run tests
-        if: ${{ matrix.arch == 'amd64' }}
+        env:
+          ROCK_NAME: ${{ steps.build.outputs.rock }}
         run: |
           set -ex
           sudo apt-get update
           sudo apt-get install -y skopeo
```

Comment:
you dont need this anymore. it's simpler and safer to use `/snap/rockcraft/current/bin/skopeo`

Reply:
Updated, thanks!